stages/02_process_data.py

```diff
@@ -0,0 +1,36 @@
+# Imports
+# -------
+import os
+import gzip
+import pathlib
+import pandas as pd
+
+files = filter( lambda item: item.is_file(), pathlib.Path('download').rglob('*') )
+
+brick_dir = pathlib.Path('brick')
+brick_dir.mkdir(exist_ok=True)
+
+for file in files:
+  file = str(file)
+  out_file = os.path.join('brick', file.split('/')[1].split('.')[0] + '.parquet')
+
+  try:
+
+      if file.endswith('xlsx'):
+          reaction_data = pd.read_excel(file)
+          reaction_data.to_parquet(out_file)
+
+      elif file.endswith('.txt.gz'):
+          with gzip.open(file, 'rb') as file_in:
+              lines = file_in.read().decode("utf-8").split('\n')
+              data = [ line.split('\t') for line in lines ]
+              headers = data.pop(0)
+              df = pd.DataFrame(data, columns=headers)
+              df.to_parquet(out_file)
+
+      else:
+          raise Exception('Unknown File Found: %s' % file)
+
+  except:
+      print ('ADRECS File Conversion Failed: %s' % file)
```

Comment:
It's easier to get rid of this outer `try-except` that way the full error can be seen. You want the error to stop the `dvc repro` process so it doesn't go any further by having a successful exit value.